### PR TITLE
Add code allowing calculation of EnsoGL stats summaries

### PR DESCRIPTION
**Pull request by akavel**
### Pull Request Description

This change adds utility code for calculating summaries from multiple samples (snapshots) of EnsoGL runtime stats values.

This internal feature is expected to be used by Enso IDE performance profiling tools, which are planned to be added in the near future.

https://www.pivotaltracker.com/story/show/181093920

#### Visuals

A demo scene named `stats` was added, showcasing how to perform calculations using the new tools. Currently, the summary calculations in the scene work only when the EnsoGL stats Monitor Panel is visible; this is planned to be improved in a future task (https://www.pivotaltracker.com/story/show/181093601).

https://user-images.githubusercontent.com/273837/152835466-b56dbd8b-b802-405f-8b43-9a97444238c5.mov

### Important Notes

 - Note: the stats aggregation code is intended to be later used in Enso IDE's main rendering loop, so it needs to have very good performance characteristics. 
     - Due to that, `Accumulator` was designed to only use simple addition arithmetic, and be constant-memory once created.

[ci no changelog needed]

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/develop/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/develop/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

File: lib/rust/ensogl/core/src/debug/stats.rs
Comment (wdanilo) [inline lib/rust/ensogl/core/src/debug/stats.rs]: Please format this code. In such complex cases, just make it multiline.

File: lib/rust/ensogl/core/src/debug/stats.rs
Comment (wdanilo) [inline lib/rust/ensogl/core/src/debug/stats.rs:349]: that's a really nice way of doing tests!

**Review by wdanilo — CHANGES_REQUESTED** [review]
The code is 100% correct, however, nothing is using this code. I have always a big trouble when merging dead code, so I'd prefer not to merge it until this PR introduces anything that actually uses this code.

**Comment by akavel** [comment]
@wdanilo I added a demo scene: it showcases the feature and provides an example of how the summarization API is intended to be used. I added a video recording of the demo scene to the PR. As mentioned by @MichaelMauderer in chat, those utilities are planned to be further used soon in https://www.pivotaltracker.com/story/show/181151201, which is waiting for them to be completed. Would the above unblock this PR for potential merging?

<blockquote><div><strong><a href="https://www.pivotaltracker.com/story/show/181151201">Integrate EnsoGL stats with profiling framework - Pivotal Tracker</a></strong></div></blockquote>

**Review by wdanilo — APPROVED** [review]
The new demo scene is amazing!

**Comment by akavel** [comment]
(*Process note:* the code-review is now accepted, therefore the PR is waiting for Review by @MichaelMauderer on Task https://www.pivotaltracker.com/story/show/181093920 before merge.)

<blockquote><div><strong><a href="https://www.pivotaltracker.com/story/show/181093920">Add code allowing calculation of EnsoGL stats summaries - Pivotal Tracker</a></strong></div></blockquote>

**Review by MichaelMauderer — APPROVED** [review]

**Pull request merged**
